### PR TITLE
ICC fixes

### PR DESCRIPTION
**Pull request by eile**
const-qualify function-local static constant arrays.

### CODE REVIEW

File: boost/network/protocol/http/client/facade.hpp [hunk 1]
```diff
@@ -131,7 +131,7 @@ class basic_client_facade {
     } else {
       if (boost::empty(content_type_headers)) {
         typedef typename char_<Tag>::type char_type;
-        static char_type content_type[] = "x-application/octet-stream";
+        static char_type* content_type = "x-application/octet-stream";
```
Comment (deanberris) [inline boost/network/protocol/http/client/facade.hpp]: I suppose a better change here would be:

```c++
static char_type const content_type[] = "...";
```

**Comment by eile** [comment]
Updated.

**Comment by deanberris** [comment]
LGTM -- thanks @eile!

**Comment by tribal-tec** [comment]
@deanberris your requested change actually does not work, at least with icc 17:

```
boost/network/protocol/http/client/facade.hpp(134): error: a value of type "const char [27]" cannot be used to initialize an entity of type "const char_type []"
          static char_type const content_type[] = "x-application/octet-stream";
```

I suggest to keep the initial code change by @eile:

```
static char_type* const content_type = "x-application/octet-stream";
```

I just verified that this compiles.

**Comment by deanberris** [comment]
@tribal-tec -- that's interesting, why isn't this an ICC bug?

If you send me a pull request I'd be happy to review and merge it.

**Pull request merged**